reasoners/lm/llama_model_v2.py

```diff
@@ -0,0 +1,282 @@
+                 hide_input: bool = True,
+                 output_log_probs: bool = False,
+                 **kwargs) -> GenerateOutput:
+        
+                                                
+        if max_length is None:
+            max_length = min(self.max_seq_len, self.model.params.max_seq_len -1)  # use LLaMA's max length if not set
+        if max_new_tokens is None:
+            max_new_tokens = max_length  # set to a large number cannot be reached
+
+        if not do_sample:
+            if temperature != 1.0 and self.local_rank == 0:  # temperature is explicitly set with do_sample=False 
+                warnings.warn('temperature is set, but do_sample=False, so temperature will be ignored.')#if this feature for you is important, you can change the warning to error
+            temperature = 0
+        
+        # unify eos_token
+        eos_token_id_input = copy.deepcopy(eos_token_id)
+        eos_token_id = []
+        if eos_token_id_input is not None:
+            if not isinstance(eos_token_id_input, list):
+                eos_token_id_input = [eos_token_id_input]
+            for token in eos_token_id_input:
+                if isinstance(token, str):
+                    tokenized = self.tokenizer.encode(token, bos=False, eos=False)
+                    if len(tokenized) != 1 and self.local_rank == 0:
+                        warnings.warn(f'the eos_token {repr(token)} is encoded into {tokenized} with length != 1, '
+                                      f'using {tokenized[-1]} as the eos_token_id')
+                    token = tokenized[-1] #this feature can be changed to eos window if user really want a special word be as eos
+                if isinstance(token, int):
+                    eos_token_id.append(token)
+                elif self.local_rank == 0:
+                    warnings.warn(f'the eos_token {repr(token)} is neither str nor int, which is ignored')
+
+        if num_return_sequences > 1:
+            assert len(inputs) == 1, 'num_return_sequences > 1 is not supported for multiple inputs'
+        
+        inputs = [i for i in inputs for _ in range(num_return_sequences)]
+
+        end_pos = torch.zeros(len(inputs)).long().cuda() - 1
+        bsz = len(inputs)
+        params = self.model.params
+        assert bsz <= params.max_batch_size, f"total batch size exceeds limit: {bsz} > {params.max_batch_size}"
+
+        prompt_tokens = [self.tokenizer.encode(x, bos=True, eos=False) for x in inputs]
+
+        min_prompt_size = min([len(t) for t in prompt_tokens])
+        max_prompt_size = max([len(t) for t in prompt_tokens])
+
+        assert max_prompt_size <= params.max_seq_len, f"prompt length exceeds limit: {max_prompt_size} > {params.max_seq_len}"
+        # total_len = min(params.max_seq_len, max_prompt_size + max_new_tokens)
+        # total_len = min(total_len, max_length)##there seems a bug, max_length can be larger than params.max_seq_len. I think we should take min
```

Comment:
We don't want to show this to users. If you are sure this is a bug please remove the comment.

Reply:
ok